### PR TITLE
[pat] Update Personal Access Token

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Implements PAT Udpate. 

* Update uses an UpdateMask to specify which fields should be updated.
* When no field is specified, the update assumes all fields should be updated.

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/14612
* Depends on https://github.com/gitpod-io/gitpod/pull/14889
* Depends on https://github.com/gitpod-io/gitpod/pull/14903
* Depends on https://github.com/gitpod-io/gitpod/pull/14901

## How to test

Unit tests

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/gitpod-db/go/personal_access_token.go [hunk 1]
```diff
@@ -180,6 +180,58 @@ func ListPersonalAccessTokensForUser(ctx context.Context, conn *gorm.DB, userID
...
+		cols = append(cols, "name")
+		update.Name = *opts.Name
+	}
+
+	if opts.Scopes != nil {
+		cols = append(cols, "scopes")
+		update.Scopes = *opts.Scopes
+	}
+
+	fmt.Println(cols)
```
Comment (andrew-farries) [inline components/gitpod-db/go/personal_access_token.go:210]: Can get rid of this.

File: components/public-api-server/pkg/apiv1/tokens.go [hunk 2]
```diff
@@ -198,23 +201,76 @@ func (s *TokensService) RegeneratePersonalAccessToken(ctx context.Context, req *
...
 
-	log.Infof("Handling UpdatePersonalAccessToken request for Token ID '%s'", tokenID.String())
-	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("gitpod.experimental.v1.TokensService.UpdatePersonalAccessToken is not implemented"))
+	toUpdate := fieldmaskpb.Intersect(mask, &updatableMask)
+	updateOpts := db.UpdatePersonalAccessTokenOpts{
+		TokenID: tokenID,
+		UserID:  userID,
+	}
+
+	fmt.Println("paths", toUpdate.GetPaths())
```
Comment (andrew-farries) [inline components/public-api-server/pkg/apiv1/tokens.go:245]: Another `Println` that we can get rid of.

**Review by andrew-farries — CHANGES_REQUESTED** [review]

File: components/public-api-server/pkg/apiv1/tokens.go (hunk 2 above)
Comment (easyCZ) [reply components/public-api-server/pkg/apiv1/tokens.go:245]: Ah, yes, thanks. I forgot to remove them after some debugging.

File: components/gitpod-db/go/personal_access_token.go (hunk 1 above)
Comment (easyCZ) [reply components/gitpod-db/go/personal_access_token.go]: Thanks. fixed now.

File: components/public-api-server/pkg/apiv1/tokens.go (hunk 2 above)
Comment (easyCZ) [reply components/public-api-server/pkg/apiv1/tokens.go]: Thanks. fixed now.

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**